File: Stack.c
#include <stdio.h>

typedef struct Stack {
    int data;
    struct Stack* next;
} Stack;

void push(Stack** top, int data) {
    Stack* new_user = (Stack*)malloc(sizeof(Stack));
    new_user->data = data;
    new_user->next = *top;
    *top = new_user;
}

int pop(Stack** top){
  int temp = (*top)->data;
  *top = (*top)->next;
  return temp;
}

int main() {
    Stack* stack = NULL; 

    push(&stack, 3);
    push(&stack, 5);
    push(&stack, 4);
    push(&stack, 2);
    int popped = pop(&stack);


    Stack* current = stack;
    while (current != NULL) {
        printf("Data: %d\n", current->data);
        current = current->next;
    }
    return 0;
}

File: Stack.py
class Node():
    def __init__(self,data):
        self.data = data
        self.next = None

class Stack():
    def __init__(self, data=None):
        if not data:
          self.head = data
        else:
          self.head = Node(data);

    def push(self,data):
        newNode = Node(data)
        newNode.next = self.head
        self.head = newNode

    def is_empty(self):
        return(not self.head)

    def pop(self):
        if self.is_empty():
          print("Empty Stack")
          return
        popped = self.head
        self.head = self.head.next
        return popped.data

    def display(self):
        temp = self.head
        while(temp):
            print(temp.data)
            temp = temp.next
    def peek(self):
        print("Empty Stack") if self.is_empty() else print(self.head.data)

    def size(self):
      num = 0
      temp = self.head
      while temp:
        num += 1
        temp = temp.next
      return num

    def clear(self):
      self.head = None